internal/planner/create.go

```diff
@@ -56,76 +56,65 @@ func (n *createNode) Kind() string { return "createNode" }
 
 func (n *createNode) Init() error { return nil }
 
-func (n *createNode) Start() error {
-	doc, err := client.NewDocFromMap(n.input, n.collection.Definition())
-	if err != nil {
-		n.err = err
-		return err
+func docIDsToSpans(ids []string, desc client.CollectionDescription) core.Spans {
+	spans := make([]core.Span, len(ids))
+	for i, id := range ids {
+		docID := base.MakeDataStoreKeyWithCollectionAndDocID(desc, id)
+		spans[i] = core.NewSpan(docID, docID.PrefixEnd())
 	}
-	n.doc = doc
-	return nil
+	return core.NewSpans(spans...)
 }
 
-// Next only returns once.
-func (n *createNode) Next() (bool, error) {
-	n.execInfo.iterations++
-
-	if n.err != nil {
-		return false, n.err
-	}
-
-	if n.returned {
-		return false, nil
-	}
-
-	if err := n.collection.Create(
-		n.p.ctx,
-		n.doc,
-	); err != nil {
-		return false, err
+func documentsToDocIDs(docs []*client.Document) []string {
+	docIDs := make([]string, len(docs))
+	for i, doc := range docs {
+		docIDs[i] = doc.ID().String()
 	}
+	return docIDs
+}
 
-	currentValue := n.documentMapping.NewDoc()
+func (n *createNode) Start() (err error) {
+	n.docs = make([]*client.Document, len(n.input))
+	defer func() { n.err = err }()
```

Comment:
todo: Please remove `n.err` completely, it is not having a positive impact on our codebase, and as far as I can tell, it has no purpose.

I assume it is a relic from before `node.Start` could return errors.

Reply:
`n.err` is checked in `Next`. It was like this before (but less consistent). Not sure it there was a good reason for this.

Reply:
removed

Reply:
cheers :)